ceph-rbd persistent volume security update

jeffvance opened the pull request.

jeffvance commented:
@adellape PTAL when you get a chance.

erinboyd commented on install_config/persistent_storage/persistent_storage_ceph_rbd.adoc:
```diff
@@ -115,13 +159,92 @@ ceph-pv                  <none>    2147483648   RWO           Available
+Even if the user and group ids are not defined in the pod spec, the resulting pod
+may have defaults defined for these ids based on its matching SCC and its project.
+The link:../../install_config/persistent_storage/pod_security_context[pod security section]
+covers storage aspcects of SCCs and defaults in greater detail.
```
typo "aspcects"

pmorie commented on install_config/persistent_storage/persistent_storage_ceph_rbd.adoc:
```diff
@@ -11,21 +11,27 @@
+volume claims (PVCs)], and with
+https://access.redhat.com/products/red-hat-ceph-storage[Ceph] is beneficial.
+
+As a very brief review, the OpenShift persistent volume (PV) framework allows administrators to
```
@adellape Is there a way to template text that is re-used?  We'll have a blurb like this for each of the volume plugins.

adellape replied:
@pmorie @jeffvance Yeah, see http://mrhaki.blogspot.com/2014/04/awesome-asciidoc-include-partial-parts.html. Example in our docs:

https://github.com/openshift/openshift-docs/blame/master/cli_reference/get_started_cli.adoc#L306

pulls in...

https://github.com/openshift/openshift-docs/blame/master/dev_guide/authentication.adoc#L118-L135

pmorie commented on install_config/persistent_storage/persistent_storage_ceph_rbd.adoc:
```diff
@@ -34,27 +40,71 @@ storage provider.
 
-- Existing Ceph volumes, to be defined in the persistent volume object
-- The `*PersistentVolume*` API
+- The ceph key (in base64) to be used as an OpenShift secret object.
```
s/(in base64)//, you don't need to do the encoding yourself anymore.

jeffvance replied:
Is base64 coding still supported?
If I do not use base64 does the ceph plugin do the bas64 conversion itself?

----- Original Message -----
> >  
> > -- Existing Ceph volumes, to be defined in the persistent volume object
> > -- The `*PersistentVolume*` API
> > +- The ceph key (in base64) to be used as an OpenShift secret object.
> 
> s/(in base64)//, you don't need to do the encoding yourself anymore.
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/openshift-docs/pull/1732/files/4059fd9b366a0c9b566e993bae7440f1b63b9808#r57236185

pmorie replied:
@jeffvance the API of the ceph volume is in terms of the name of the secret to refer to, the ceph plugin knows nothing about how secret data is encoded when serialized.

pmorie commented on install_config/persistent_storage/persistent_storage_ceph_rbd.adoc:
```diff
@@ -34,27 +40,71 @@ storage provider.
-- The `*PersistentVolume*` API
+- The ceph key (in base64) to be used as an OpenShift secret object.
+- The ceph image name.
+- The file system type on top of the block storage (e.g., ext4).
```
@adellape are fs types usually monospaced in the docs?

adellape replied:
So far we have not, and looking around in RHEL documentation, it looks like _most_ of the time it is not. So I guess we can leave it plain.

pmorie commented on install_config/persistent_storage/persistent_storage_ceph_rbd.adoc:
```diff
@@ -34,27 +40,71 @@ storage provider.
+[[creating-ceph-secret]]
+=== Creating the Ceph Secret
+
+In a secret definition, the storage administrator must define the admin authorization key,
```
we need to expand this section to either cover or reference doc for `oc create secret`

pmorie replied:
Either way, I think the doc should basically not include any mention of base64.  You can create a secret from a literal value like `oc create secret ceph-secret --from-literal=key=valueOfKey`, no base64 encoding required.

jeffvance replied:
# openshift version
openshift v3.1.1.6
kubernetes v1.1.0-origin-1107-g4c8e6f4
etcd 2.1.2

-- creating pod with non-base64 secret results in the pod failing with this mount error:
2s		6	{kubelet rhel7.2-dev}			FailedMount	Unable to mount volumes for pod "ceph-pod1_default": fork/exec /usr/bin/rbd: invalid argument

Changing nothing else except for base64 encoding the secret works.
@pmorie Perhaps this enhancement was added to a newer verison of openshift?

pmorie replied:
My bad, @jeffvance , the command is `oc create secret generic`.

pmorie commented on install_config/persistent_storage/persistent_storage_ceph_rbd.adoc:
```diff
@@ -34,27 +40,71 @@ storage provider.
+----
+====
+
+Verify that the secret was created:
```
From here down is okay re: secrets.

pmorie commented on install_config/persistent_storage/persistent_storage_ceph_rbd.adoc:
```diff
@@ -64,35 +114,36 @@ it in OpenShift:
+single user (non-shared storage).
+<4> This defines the volume type being used, in this case the *rbd* plug-in.
+<5> This is an array of Ceph monitor ip addresses and ports.
+<6> This is the Ceph secret, define above, used to create a secure connection
```
s/define/defined/

pmorie commented on install_config/persistent_storage/persistent_storage_ceph_rbd.adoc:
```diff
@@ -115,13 +166,92 @@ ceph-pv                  <none>    2147483648   RWO           Available
+A significant difference between _shared_ volumes (such as NFS or GlusterFS) and
+_block_ volumes (such as Ceph-RBD, iSCSI, most cloud storage) is that the UID
+and/or group IDs defined in the pod definition (or in the docker image) are applied
+to the *_target_* (physical) storage. This is referred to as *_taking over_* the
```
Please say 'managing ownership of the block device' here and in all other similar docs.

pmorie commented on install_config/persistent_storage/persistent_storage_ceph_rbd.adoc:
```diff
@@ -115,13 +166,92 @@ ceph-pv                  <none>    2147483648   RWO           Available
+
+
+
+A pod can define a group ID using `*fsGroup*` under the pod's global
```
`A pod defines the group that should own the volume...`

jeffvance commented:
@pmorie Thanks for your feedback! I've incorporated it in the most recent push.

jeffvance commented:
@adellape Thanks for the review. I've pushed the latest changes.

adellape commented on install_config/persistent_storage/persistent_storage_ceph_rbd.adoc:
```diff
@@ -115,13 +167,87 @@ ceph-pv                  <none>    2147483648   RWO           Available
+<2> This claim will look for PVs offering *2Gi* or greater capacity.
+====
+
+Save the definition to a file, for example *ceph-claim.yaml_*, and create the
```
Missing an underscore in `*ceph-claim.yaml_*`

adellape commented on install_config/persistent_storage/persistent_storage_ceph_rbd.adoc:
```diff
@@ -115,13 +167,87 @@ ceph-pv                  <none>    2147483648   RWO           Available
+====
+
+[[ceph-volume-security]]
+=== Volume Security
```
Can you bring this up one level?

adellape commented on install_config/persistent_storage/persistent_storage_ceph_rbd.adoc:
```diff
@@ -115,13 +167,87 @@ ceph-pv                  <none>    2147483648   RWO           Available
+A significant difference between *shared* volumes (such as NFS and GlusterFS) versus
+*block* volumes (such as Ceph-RBD, iSCSI, and most cloud storage) is that the user and
+group IDs defined in the pod definition (or in the docker image) are applied to the
+*_target_* (physical) storage. This is referred to as *_managing ownership* of the
```
Missing a closing underscore on `*_managing ownership*`

adellape commented:
@jeffvance Sorry, hadn't really gotten to this one other than replying to Paul the other day. Added a couple comments inline. After those are fixed and squashed, I'll merge this and we (docs) will follow-up w/ any further edits. Thanks!

jeffvance commented:
Thanks Alex. Pushed up the latest based on your feedback.

----- Original Message -----
> @jeffvance Sorry, hadn't really gotten to this one other than replying to
> Paul the other day. Added a couple comments inline. After those are fixed
> and squashed, I'll merge this and we (docs) will follow-up w/ any further
> edits. Thanks!
> 
> ---
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/openshift-docs/pull/1732#issuecomment-204043213

erinboyd commented:
@adellape is this ready to merge?

jeffvance commented:
@adellape PTAL again. It's been ready for a while I believe.

adellape commented:
@jeffvance Sorry, missed it. Merging!
